File: resturant-backend/src/main/java/com/spring/boot/resturantbackend/models/contact_info/ContactInfo.java
package com.spring.boot.resturantbackend.models.contact_info;

import com.spring.boot.resturantbackend.models.BaseEntity;
import com.spring.boot.resturantbackend.models.security.Account;
import com.spring.boot.resturantbackend.utils.enums.FilterContactInfo;
import jakarta.persistence.*;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Entity
@Table(schema = "hr")
@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter
public class ContactInfo extends BaseEntity<String> {
    @Column(nullable = false)
    @NotBlank(message = "not_empty.name")
    @Size(min = 7, max = 100, message = "size.name")
    private String name;
    @Column(nullable = false)
    @Email(message = "not_valid.email")
    @NotBlank(message = "not_empty.email")
    private String email;
    @Column(nullable = false)
    @NotBlank(message = "not_empty.subject")
    private String subject;
    @Column(nullable = false)
    @NotBlank(message = "not_empty.message")
    private String message;
    @Column(nullable = false)
    @NotNull(message = "not_empty.status")
    @Enumerated(EnumType.STRING)
    private FilterContactInfo status;
    @ManyToOne(fetch = FetchType.LAZY)
    private Account senderAccount;
    @ManyToOne(fetch = FetchType.LAZY)
    private Account receivedAccount;
    @OneToMany(mappedBy = "contactInfo", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Comment> comment;
}

File: front-end-903/src/model/contact-info.ts
import {CommentContactInfo} from './commentContactInfo';

export enum FilterContactInfo {
  ALL = 'ALL',
  NOT_REPLIED = 'NOT_REPLIED',
  REPLIED = 'REPLIED'
}

export class ContactInfo {
  id?: number | null;
  status?: FilterContactInfo | null;
  comment?: CommentContactInfo[] | null;

  name: string;
  email: string;
  subject: string;
  message: string;
  accountId: number;

  constructor({
                name,
                email,
                subject,
                message,
                id = null,
                status = null,
                comment = null,
                accountId = null
              }: {
    name: string;
    email: string;
    subject: string;
    message: string;
    id?: number | null;
    status?: FilterContactInfo | null;
    comment?: CommentContactInfo[] | null;
    accountId?: number | null;
  }) {
    this.name = name;
    this.email = email;
    this.subject = subject;
    this.message = message;
    this.id = id;
    this.status = status;
    this.comment = comment;
    this.accountId = accountId;
  }
}
